lib/ai/embeddingprocessor.go

```diff
@@ -217,19 +213,35 @@ func (e *EmbeddingProcessor) process(ctx context.Context) {
 	e.log.Debugf("embedding processor started")
 	defer e.log.Debugf("embedding processor finished")
 
-	embeddingsStream := e.embeddingSrv.GetEmbeddings(ctx, types.KindNode)
-	nodesStream := e.nodeSrv.GetNodeStream(ctx, defaults.Namespace)
+	embeddingsStream := e.embeddingSrv.GetAllEmbeddings(ctx)
+	unifiedResources, err := e.nodeSrv.GetUnifiedResources(ctx)
+	if err != nil {
+		e.log.Debugf("embedding processor failed with error: %v", err)
+		return
+	}
+
+	idx := 0
+	resourceStream := stream.Func(func() (types.Resource, error) {
+		if idx == len(unifiedResources) {
+			return nil, io.EOF
+		}
+
+		resource := unifiedResources[idx].(types.Resource)
```

Comment:
I know this will never panic, but just in case if impossible happens, can you add the type check?

Reply:
sure